app/src/main/java/org/oppia/android/app/options/AppLanguageFragmentPresenter.kt

```diff
@@ -25,25 +31,58 @@ class AppLanguageFragmentPresenter @Inject constructor(private val fragment: Fra
       /* attachToRoot= */ false
     )
     this.prefSummaryValue = prefSummaryValue
-    languageSelectionAdapter = LanguageSelectionAdapter(prefKey) {
-      updateAppLanguage(it)
-    }
+    languageSelectionViewModel = getLanguageSelectionViewModel()
+    binding.viewModel = languageSelectionViewModel
+    languageSelectionViewModel.selectedLanguage.value = prefSummaryValue
     binding.languageRecyclerView.apply {
-      adapter = languageSelectionAdapter
+      adapter = createRecyclerViewAdapter()
     }
 
     binding.appLanguageToolbar?.setNavigationOnClickListener {
-      val message = languageSelectionAdapter.getSelectedLanguage()
+      val message = languageSelectionViewModel.selectedLanguage.value
       val intent = Intent()
       intent.putExtra(MESSAGE_APP_LANGUAGE_ARGUMENT_KEY, message)
       (fragment.activity as AppLanguageActivity).setResult(REQUEST_CODE_APP_LANGUAGE, intent)
       (fragment.activity as AppLanguageActivity).finish()
     }
 
-    createAdapter()
     return binding.root
   }
 
+  fun getLanguageSelected(): String? {
+    return languageSelectionViewModel.selectedLanguage.value
+  }
+
+  private fun createRecyclerViewAdapter(): BindableAdapter<LanguageItemViewModel> {
+    return BindableAdapter.MultiTypeBuilder
+      .newBuilder<LanguageItemViewModel, ViewType> {
+        ViewType.VIEW_TYPE_LANGUAGE
+      }
+      .registerViewDataBinder(
+        viewType = ViewType.VIEW_TYPE_LANGUAGE,
+        inflateDataBinding = LanguageItemsBinding::inflate,
+        setViewModel = this::bindSkillView,
+        transformViewModel = { it }
+      ).build()
+  }
+
+  private fun bindSkillView(
+    binding: LanguageItemsBinding,
+    model: LanguageItemViewModel
+  ) {
+    binding.viewModel = model
+    binding.radioContainer.setOnClickListener {
+      languageSelectionViewModel.selectedLanguage.value = model.language
+      updateAppLanguage(model.language)
+    }
+    languageSelectionViewModel.selectedLanguage.observe(
+      fragment,
+      Observer {
+        binding.isChecked = model.language == it
+      }
+    )
```

Comment:
Could we do these through databinding, instead? It might be a bit cleaner that way (e.g. defining an onClick for the ratio container, or binding the checkbox's state on a transformation of the live data). In general, we should try to keep to a single, trivial binding call to get the most leverage out of BindableAdapter.

Ditto for audio language.

Reply:
@BenHenning I'm using an observer here to update other languages to unselect them

Reply:
Does the following work:

Kotlin changes in LanguageItemViewModel (you'll need to pass in the selectedLanguage as a parameter):
```kotlin
val isLanguageSelected: LiveData<Boolean> by lazy {
  Transformations.map(selectedLanguage) { it == language }
}
```

And then in XML you would bind to ``isLanguageSelected`` instead of ``isChecked`` (databinding can directly bind to a LiveData & it'll automatically add a correctly-scoped observer to it).

Reply:
Also for the onClick, you'd probably need to introduce a listener that's passed down to the view model which is called in onClick(). That listener will likely correspond to the outer fragment or activity, then you'll call down to the presenter to update the shared live data.

Reply:
@BenHenning I'm facing this error for the code you've suggested 

```
Type inference failed: fun <X : Any!, Y : Any!> map(p0: LiveData<X!>, p1: Function<X!, Y!>): LiveData<Y!>
cannot be applied to
(String,() -> Boolean)
```

Reply:
@FareesHussain you may need to use a separate function rather than using a lambda (such as [here](https://github.com/oppia/oppia-android/blob/141511329ea0249ff225a469c70658c3b2123238/app/src/main/java/org/oppia/android/app/profileprogress/ProfileProgressViewModel.kt#L74)). My code was meant to be a sample, not compile-ready code. You may need to make additional changes to make it work in your particular circumstance since it was meant to convey an idea.

Reply:
@BenHenning I've made these change but I'm unable to check and uncheck the radio button here

Reply:
also `isLanguageSelected` doesn't work here

Reply:
@FareesHussain can you detail what you've tried & confirmed does & does not work about the new flow that's been implemented? The more specific, the better (e.g. what specific steps have you repro'd for the failure, what specific parts of the flow have you confirmed working correctly in that failure scenario & what specifically seems wrong from your debugging).

Reply:
@BenHenning I'm unable to check and uncheck the radio button it is not responding with `isLanguageSelected` 
<img src="https://user-images.githubusercontent.com/54740946/105269573-3370b080-5bba-11eb-824d-e3510138e013.png" width="33%"/>
Reset works fine it's just the Radio button which is not functional

Reply:
also `isLanguageSelected` gives null on debugging

Reply:
also the added parameter `selectedLanguage` does'nt get updated

Reply:
How are you debugging the value? Per Kotlin, that value cannot be null, so it's quite concerning if you're observing that. I don't know specifically what's broken here--you'll need to continue digging into it. Ensuring the dataflow pipeline is hooked up correctly is one of the main challenges with this issue, and I unfortunately don't have any tips beyond what I've shared so far. Feel free to keep asking questions as you continue to dig into it--I might have thoughts once more context around the issue is discovered.

The one thing I'll mention is that this should be possible. Checkboxes are supposed to work with LiveData, and it seems like the way you're setting it up is correct. I suspect there's a subtle part missing or incorrect here that needs to be correct, and without digging into it myself I won't be able to figure out what that is.

Reply:
I stepped through the debugger a bit and noticed that we change the selected language in [this view model here](https://github.com/oppia/oppia-android/pull/2392/files#diff-54e88ee346fe26a3b0ace7c54558ece962b0fa1d4aff20177fdf4e4bd78c4070R20), but it looks like the change doesn't propagate to [this view model here](https://github.com/oppia/oppia-android/pull/2392/files#diff-c2847766e4306b3b4ce77bc78349621ef3cd1f9d2323ca8211c8e57b5168dc26R13), which is what is bound to the radio button.

Have you tried adding a function & doing some processing inside the `LanguageItemViewModel`, like the `SelectionInteractionContentViewModel` [here](https://github.com/oppia/oppia-android/blob/c7481dd7777d6ee6a07113c5ec56002e2629666a/app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/SelectionInteractionContentViewModel.kt#L15). So then the layouts [here](https://github.com/oppia/oppia-android/pull/2392/files#diff-eeb58f7d33e61d343f04418427490025964091734bc7a1e1272c9dcc36a0cbbcR17) would look like `viewModel.handleClick()` and `handleClick()` would call `languageRadioButtonListener.onLanguageSelection()`